components/zcash_address/src/kind/unified.rs

```diff
@@ -275,14 +232,10 @@ impl TryFrom<(&str, &[u8])> for Address {
             result.push(read_receiver(&mut cursor)?);
```

Comment:
I don't understand how this can work. `read_receiver` will be monomorphized to a particular `R: Receiver` but I see nothing that specalizes `R` to a particular type at this call site. How does the compiler know what `R` to pick? This is much more convoluted than the `enum`-based approach. If `R ~ Self::Receiver` then there's no need for `read_receiver` to be polymorphic.

Reply:
Okay, so this is because inner functions don't properly close over the type parameters of the outer function.